daemon/sonica-daemon.py

```diff
@@ -169,27 +177,71 @@ def Library(self, request, context):
         )
 
 
-
-def start_server(sonica):
+def start_server(sonica, port):
     server = grpc.server(ThreadPoolExecutor(max_workers = 4))
     add_SonicaServicer_to_server(sonica, server)
 
-    location = 'localhost:7700'
+    location = f'localhost:{port}'
     server.add_insecure_port(location)
     server.start()
     print(f'Server started on {location}')
     server.wait_for_termination()
 
-def main(deez_arl : str = None, dir : str = 'music'):
-    library = Library(dir)
+def opts_to_map(opts: List[str]):
+    m = {}
+    for s in opts:
+        k, v = s.split("=")
+        engine, name = k.split(":")
+        if not engine in m:
+            m[engine] = {}
+        m[engine][name] = v
+    return m
+
+def load_engines(dirs : List[str], opts, library):
+    engines = []
+    paths = [ os.path.abspath(os.path.expanduser(d)) for d in dirs ]
+    for path in paths:
+        if not os.path.exists(path):
+            continue
+
+        sys.path.insert(1, str(path))
+
+        for item in os.listdir(path):
+            if ".py" in item and item != "engine.py":
+
+                # Name generation
+                lib = item.replace(".py", "") # test_engine.py => test_engine
+                name = lib.replace("_engine", "") # test_engine => test
+                eclass = lib.title().replace("_", "") # test_engine => TestEngine
+                print(f"Loading {eclass} from {item}")
+
+                # Loading the engine
+                eopts = opts[name] if name in opts else {}
+                e = getattr(__import__(lib), eclass)(library, eopts)
+                engines.append(e)
+
+    return engines
+
+
+def main(
+        port : int = 7700,
+        music_dir : str = 'music',
+        engines_dir : List[str] = ['engines', '~/.config/sonicad/plugins'],
+        engine_opt : List[str] = typer.Option([], "--engine-opt", "-o", help="Additional options to parse on to an engine ie. engine:option=value")
+    ):
+    if not os.path.exists(music_dir):
+        print(f"Given music dir does not exist: »{music_dir}«", file=sys.stderr)
+        exit(1)
```

Comment:
Could just create the directory and give an alert about it instead

Reply:
I'd personally prefer it to tell me something is wrong, since i think the general usecase will be pointing it at some location set up to be a music folder.

Reply:
That's fair. Alternatively, it could be an option, something like `--force-music-dir`